File: go/models/cancel_result.go
package models

// CancelDetail represents the result detail for a single cancel operation.
type CancelDetail struct {
	CargoKey         string `xml:"cargoKey"`
	InvoiceKey       string `xml:"invoiceKey"`
	JobId            int    `xml:"jobId"`
	DocId            string `xml:"docId"`
	OperationCode    int    `xml:"operationCode"`
	OperationMessage string `xml:"operationMessage"`
	OperationStatus  string `xml:"operationStatus"`
	ErrCode          int    `xml:"errCode"`
	ErrMessage       string `xml:"errMessage"`
}

// CancelResult represents the response from cancelShipment.
type CancelResult struct {
	OutFlag      int            `xml:"outFlag"`
	OutResult    string         `xml:"outResult"`
	SenderCustId int            `xml:"senderCustId"`
	Count        int            `xml:"count"`
	Details      []CancelDetail `xml:"cancelShipmentDetailVO"`
}

// IsSuccess returns true if the operation was successful.
func (r *CancelResult) IsSuccess() bool {
	return r.OutFlag == 0
}

File: dotnet/Models/CancelResult.cs
namespace YurticiKargo.Models;

public class CancelDetail
{
    public string CargoKey { get; set; } = string.Empty;
    public string InvoiceKey { get; set; } = string.Empty;
    public long JobId { get; set; }
    public string DocId { get; set; } = string.Empty;
    public int OperationCode { get; set; }
    public string OperationMessage { get; set; } = string.Empty;
    public string OperationStatus { get; set; } = string.Empty;
    public int ErrCode { get; set; }
    public string ErrMessage { get; set; } = string.Empty;
}

public class CancelResult
{
    public int OutFlag { get; set; }
    public string OutResult { get; set; } = string.Empty;
    public long SenderCustId { get; set; }
    public int Count { get; set; }
    public List<CancelDetail> Details { get; set; } = new();

    public bool IsSuccess() => OutFlag == 0;
}
